### PR TITLE
Media post list layout

### PR DESCRIPTION
**Pull request by jmbarne3**
**Description**
Provides a layout with a group by `subterm` function. This basically allows you to define a parent taxonomy term, then group all the posts by child terms of that parent. 

**Motivation and Context**
We're going to be using this layout to display forms grouped by some sort of hierarchical taxonomy. 

**How Has This Been Tested?**
Changes are available for review in dev. 

**Types of changes**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

**Checklist:**

- [x] My code follows the code style of this project.
- [ ] My change requires an update to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: plugin-filters/finaid-post-list-layout.php [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+ */
+function ucf_post_list_display_media_before( $content, $posts, $atts ) {
+	ob_start();
+?>
+	<div class="ucf-post-list ucf-post-list-media" id="post-list-<?php echo $atts['list_id']; ?>">
+<?php
+	return ob_get_clean();
+}
+
+// add_filter( 'ucf_post_list_display_media_before', array( __NAMESPACE__ . "\ucf_post_list_display_media_before" ), 10, 3 );
```
Comment (cjg89) [inline plugin-filters/finaid-post-list-layout.php]: Filter addition is commented out here?

File: plugin-filters/finaid-post-list-layout.php [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+/**
+ * Function for displaying the items of the media post list
+ * @author Jim Barnes
+ * @since 1.0.0
+ * @param string $content The content passed to the filter
+ * @param array $posts The posts
+ * @param array $atts The shortcode arguments
+ * @return string
+ */
+function ucf_post_list_display_media( $content, $posts, $atts ) {
```
Comment (cjg89) [inline plugin-filters/finaid-post-list-layout.php]: Need to add a check at the top of this function that ensures `$posts` is an array, e.g.
`if ( ! is_array( $posts ) && $posts !== false ) { $posts = array( $posts ); }`

Logic in this function should also check to ensure `$posts` is not false-y before attempting to loop through it

File: plugin-filters/finaid-post-list-layout.php (hunk 1 above)
Comment (jmbarne3) [reply plugin-filters/finaid-post-list-layout.php]: Pew pew.

File: plugin-filters/finaid-post-list-layout.php [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+	// If we're grouping by subterm, go ahead and do it
+	list($grouped, $posts) = $group_by_subterm ? ucf_post_list_grouped_groupby_subterm( $posts, $atts ) : array(false, $posts);
+
+	ob_start();
+
+	if ( $grouped ) :
+		foreach( $posts as $term ) :
+?>
+	<h2 class="heading-underline"><?php echo $term['term_name']; ?></h2>
+	<ul class="mb-2 list-unstyled">
```
Comment (cadie) [inline plugin-filters/finaid-post-list-layout.php]: What do you think about increasing this margin-bottom to `mb-4`? The extra space helps separate out the groupings visually.

File: plugin-filters/finaid-post-list-layout.php (hunk 2 above)
Comment (jmbarne3) [reply plugin-filters/finaid-post-list-layout.php]: Added, along with a count check before the grouping.

File: plugin-filters/finaid-post-list-layout.php (hunk 3 above)
Comment (jmbarne3) [reply plugin-filters/finaid-post-list-layout.php]: Updated!

**Pull request merged**